src/app/api/courses/[courseId]/route.ts

```diff
@@ -0,0 +1,37 @@
+import { getCourseInfo } from '@services/archive'
+import { validateRights } from '@services/tkoUserService'
+import { NextResponse } from 'next/server'
+
+export const GET = async (
+  req: Request,
+  {
+    params
+  }: {
+    params: { courseId: string }
+  }
+) => {
+  try {
+    const isRights = await validateRights('access')
+    if (!isRights) {
+      return NextResponse.json(
+        { error: '401 Unauthorized' },
```

Comment:
These are repeated often in the codebase, is there a utility to just say `401` or could we create one for ourselves? The `!isRights -> 401` pattern is also very common, is there middleware or a function we could use to reduce the repetition (and allow easier changes)?

Reply:
Only repetition that we could get out of this is move the `NextResponse.json(...)` into own function, but I think that would not reduce repetition, only add more complexity. This `NextResponse` is only used in APIs and Next.js needs that to be returned out of the route handler function.

Reply:
This will be revisited when NextAuth upcoming version becomes more stable as handling session in route handlers will be different than this is currently having.